cmake/podspec_cmake.rb

```diff
@@ -0,0 +1,494 @@
+# +req_subspecs+:: Which subspecs to include. If empty, all subspecs are
+#   included (which corresponds to CocoaPods behavior. The default_subspec
+#   property is not handled.
+def process(podspec_file, cmake_file, *req_subspecs)
+  root_dir = Pathname.new(__FILE__).expand_path().dirname().dirname()
+  path_list = Pod::Sandbox::PathList.new(root_dir)
+
+  spec = Pod::Spec.from_file(podspec_file)
+
+  writer = Writer.new()
+  writer.append <<~EOF
+  # This file was generated by #{File.basename(__FILE__)}
+  # from #{File.basename(podspec_file)}.
+  # Do not edit!
+  EOF
+
+  cmake_binary_dir = File.expand_path(File.dirname(cmake_file))
+
+  gen = CMakeGenerator.new(spec, path_list, cmake_binary_dir)
+  gen.add_framework(spec)
+
+  req_subspecs = normalize_requested_subspecs(spec, req_subspecs)
+  req_subspecs = resolve_subspec_deps(spec, req_subspecs)
+
+  spec.subspecs.each do |subspec|
+    if req_subspecs.include?(subspec.name)
+      gen.add_framework(subspec)
+    end
+  end
+
+  gen.target.commands.each do |command|
+    writer.write(command)
+  end
+
+  File.open(cmake_file, 'w') do |fd|
+    fd.write(writer.result)
+  end
+end
+
+# Translates the (possibly empty) list of requested subspecs into the list of
+# subspecs to actually include. If +req_subspecs+ is empty, returns all
+# subspecs. If non-empty, all subspecs are returned as qualified names, e.g.
+# "Logger" may become "GoogleUtilities/Logger".
+def normalize_requested_subspecs(spec, req_subspecs)
+  subspecs = spec.subspecs
+  if req_subspecs.empty?
+    return subspecs.map { |s| s.name }
+  else
+    return req_subspecs.map do |name|
+      if name.include?(?/)
+        name
+      else
+        "#{spec.name}/#{name}"
+      end
+    end
+  end
+end
+
+# Expands the list of requested subspecs to include any dependencies within the
+# same root subspec. For example, if +req_subspecs+ where
+#
+#   +["GoogleUtilties/Logger"]+,
+#
+# the result would be
+#
+#   +["GoogleUtilties/Logger", "GoogleUtilities/Environment"]+
+#
+# because Logger depends upon Environment within the same root spec.
+def resolve_subspec_deps(spec, req_subspecs)
+  prefix = spec.name + '/'
+
+  result = Set.new()
+  while !req_subspecs.empty?
+    req = req_subspecs.pop
+    result.add(req)
+
+    subspec = spec.subspec_by_name(req)
+    subspec.dependencies(PLATFORM).each do |dep|
+      if dep.name.start_with?(prefix) && !result.include?(dep.name)
+        req_subspecs.push(dep.name)
+      end
+    end
+  end
+
+  return result.to_a.sort
+end
+
+# Writes CMake commands out to textual form, taking care of line wrapping.
+class Writer
+  def initialize()
+    @last_command = nil
+    @result = ""
+  end
+
+  attr_reader :result
+
+  def write(command)
+    if command.skip?
+      return
+    end
+
+    if command.name != @last_command
+      @result << "\n"
+    end
+    @last_command = command.name
+
+    single = "#{command.name}(#{command.rest.join(' ')})\n"
+    if single.size < 80
```

Comment:
It's not generally considered necessary to follow any form of style guidelines for generated code. You could simplify this function by eliminating the stylistic parts. (Though it *does* make the generated code easier to read, which is sometimes helpful.)

Optional.

Reply:
Given that this is potentially tens of fully qualified pathnames as arguments I think trying to put one of those per line is worth the extra eight lines here. CMake doesn't care but I do when trying to manually verify that everything is being included as intended.